eigen/client.go

```diff
@@ -98,66 +75,98 @@ func (e *Client) GetProof(ctx context.Context, id da.ID) (da.Proof, error) {
 	return nil, nil
 }
 
-type ID struct {
-	BlobIndex       uint32
-	BatchHeaderHash []byte
+// PutBlob encodes and writes a blob to EigenDA, waiting for it to be confirmed
+// before returning. This function is resiliant to transient failures and
+// timeouts.
+func (c *Client) PutBlob(ctx context.Context, data []byte) (*grpcdisperser.BlobInfo, error) {
+	resultChan, errorChan := c.PutBlobAsync(ctx, data)
+	select { // no timeout here because we depend on the configured timeout in PutBlobAsync
+	case result := <-resultChan:
+		return result, nil
+	case err := <-errorChan:
+		return nil, err
+	}
 }
 
-func (e *Client) disperseBlob(ctx context.Context, txData []byte) (*BlobInfo, error) {
-	fmt.Println("Attempting to disperse blob to EigenDA")
+func (c *Client) PutBlobAsync(ctx context.Context, data []byte) (resultChan chan *grpcdisperser.BlobInfo, errChan chan error) {
+	resultChan = make(chan *grpcdisperser.BlobInfo, 1)
+	errChan = make(chan error, 1)
+	go c.putBlob(ctx, data, resultChan, errChan)
+	return
+}
```

Comment:
Is this implemented in this async manner because the confirmation and response from eigenDA with blob ID takes about 5 mins?

Reply:
Sync works just as well actually (without a go-routine or channels)
This pattern is generally used to replicate async/await. So
```go
	go c.putBlob(ctx, data, resultChan, errChan)
	// do some work while we wait for blob
	result := <- resultChan
```
Which we are not using here. 
Original code source: https://github.com/Layr-Labs/eigenda/blob/12f1a6f6c8951cda36b93f1060e8ce441b06b6ec/api/clients/eigenda_client.go#L99